[Bug] Nil pointer error when running locally

### Thank you for the bug report

- [X] I am using the latest version of the `OrgNote` app.
- [X] I already checked other related issues
- [X] I have read the [WIKI](https://github.com/Artawower/orgnote/wiki)

### Bug description and steps to reproduce

# Steps to reproduce
- Clone the orgnote-backend repo
- Make sure `ACCESS_CHECK_URL` is unset
- Build and run the development image with `docker compose -f docker-compose.db.yaml -f docker-compose.dev.yaml up --build command

# .env file for reproducing
```
DEBUG=true
MONGO_USERNAME=admin
MONGO_PASSWORD=admin
BACKEND_SCHEMA=http
BACKEND_DOMAIN=localhost
BACKEND_PORT=8000
```

# Error output
```
{"level":"warn","time":"2024-12-14T16:53:50Z","message":"Github OAuth is not configured"}
{"level":"info","time":"2024-12-14T16:53:50Z","message":"Config: (configs.Config) {\n AppAddress: (string) (len=12) \"0.0.0.0:8000\",\n MongoURI: (string) (len=41) \"mongodb://admin:admin@orgnote_mongo:27017\",\n Debug: (bool) true,\n MediaPath: (string) (len=7) \"./media\",\n GithubID: (string) \"\",\n GithubSecret: (string) \"\",\n BackendSchema: (string) (len=4) \"http\",\n BackendPort: (string) (len=4) \"8000\",\n BackendDomain: (string) (len=9) \"localhost\",\n ClientAddress: (string) (len=12) \"0.0.0.0:8000\",\n MobileAppName: (string) (len=7) \"orgnote\",\n AccessCheckerURL: (*string)(<nil>),\n AccessCheckToken: (*string)(<nil>),\n AccessTokenCacheLifeTime: (int) 60,\n MaximumFileSize: (int) 10485760,\n GithubClientOwner: (string) (len=9) \"artawower\",\n GithubClientRepoName: (string) (len=14) \"orgnote-client\"\n}\n"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa21a2e]

goroutine 1 [running]:
orgnote/app/infrastructure.NewSubscription({{0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0}, 0x0, 0x0, 0x14f7cb0, 0x3c)
        /app/app/infrastructure/subscription.go:158 +0x4e
main.main()
        /app/app/main.go:69 +0x4ea
[16:53:50] Process Exit with Code: 2
```

### Additional context

_No response_

--- a/Dockerfile
+++ b/Dockerfile
@@ -12,6 +12,6 @@ RUN go install github.com/deepmap/oapi-codegen/cmd/oapi-codegen@latest
 COPY generate-subscription-api.sh generate-subscription-api.sh
 RUN sh generate-subscription-api.sh
 RUN go build -o orgnote app/main.go
-RUN go install -tags 'mongodb' github.com/golang-migrate/migrate/v4/cmd/migrate@latest
+RUN go install -tags 'mongodb' github.com/golang-migrate/migrate/v4/cmd/migrate@v4.17.1
 
 ENTRYPOINT ["./entrypoint.sh"]

--- a/Dockerfile-dev
+++ b/Dockerfile-dev
@@ -11,6 +11,6 @@ RUN go mod download && go mod verify
 RUN curl -sSfL https://raw.githubusercontent.com/cosmtrek/air/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
 RUN go install github.com/swaggo/swag/cmd/swag@latest
 RUN go install github.com/deepmap/oapi-codegen/cmd/oapi-codegen@latest 
-RUN go install -tags 'mongodb' github.com/golang-migrate/migrate/v4/cmd/migrate@latest
+RUN go install -tags 'mongodb' github.com/golang-migrate/migrate/v4/cmd/migrate@v4.17.1
 
 CMD ["air"]

--- a/app/configs/config.go
+++ b/app/configs/config.go
@@ -74,9 +74,7 @@ func NewConfig() Config {
 		log.Fatal().Msg("BACKEND_DOMAIN or BACKEND_SCHEMA is not set")
 	}
 
-	var accessCheckerURL *string
 	envAccessCheckURL := os.Getenv("ACCESS_CHECK_URL")
-	accessCheckerURL = &envAccessCheckURL
 
 	var accessCheckToken *string
 	if envAccessCheckToken := os.Getenv("ACCESS_CHECK_TOKEN"); envAccessCheckToken != "" {
@@ -117,7 +115,7 @@ func NewConfig() Config {
 		BackendSchema:            backendSchema,
 		BackendDomain:            backendDomain,
 		BackendPort:              backendPort,
-		AccessCheckerURL:         accessCheckerURL,
+		AccessCheckerURL:         &envAccessCheckURL,
 		AccessCheckToken:         accessCheckToken,
 		AccessTokenCacheLifeTime: accessTokenCacheLifeTime,
 		MaximumFileSize:          maximumFileSize,

--- a/app/configs/config.go
+++ b/app/configs/config.go
@@ -75,9 +75,8 @@ func NewConfig() Config {
 	}
 
 	var accessCheckerURL *string
-	if envAccessCheckURL := os.Getenv("ACCESS_CHECK_URL"); envAccessCheckURL != "" {
-		accessCheckerURL = &envAccessCheckURL
-	}
+	envAccessCheckURL := os.Getenv("ACCESS_CHECK_URL")
+	accessCheckerURL = &envAccessCheckURL
 
 	var accessCheckToken *string
 	if envAccessCheckToken := os.Getenv("ACCESS_CHECK_TOKEN"); envAccessCheckToken != "" {

